modules/ximgproc/perf/perf_edgepreserving_filter.cpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html.
+#include "perf_precomp.hpp"
+
+namespace opencv_test
+{
+namespace
+{
+
+/* 1. Define parameter type and test fixture */
+typedef tuple<int, double> RGFTestParam;
+typedef TestBaseWithParam<RGFTestParam> EdgepreservingFilterTest;
+
+/* 2. Declare the testsuite */
+PERF_TEST_P(EdgepreservingFilterTest, perf,
+            Combine(Values(-20, 0, 10), Values(-100, 0, 20)))
+{
+	/* 3. Get actual test parameters */
+	RGFTestParam params = GetParam();
+	int kernelSize = get<0>(params);
+	double threshold = get<1>(params);
+
+	/* 4. Allocate and initialize arguments for tested function */
+	std::string filename = getDataPath("samples/data/corridor.jpg");
```

Comment:
Is it considered good practice to use [https://github.com/opencv/opencv_extra/blob/master/testdata/perf/320x260.png](https://github.com/opencv/opencv_extra/blob/master/testdata/perf/320x260.png) instead of creating my own image? 

Thus, no patch would be needed.

Reply:
Reusing existed test images is a good practice.